### ISSUE TITLE
NLL migrate mode: issues two errors for move out of single static item

### ISSUE THREAD

**Issue by pnkfelix** [rust-lang/rust#55309]
This code ([play](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=ae99527dd0a565507c268a94ccb9809b))

```rust
struct S;
pub fn main() {
    let _y = { static x: S = S; x };
}
```

issues the following diagnostics under migrate mode:

```
error[E0507]: cannot move out of static item
 --> src/main.rs:3:33
  |
3 |     let _y = { static x: S = S; x };
  |                                 ^
  |                                 |
  |                                 cannot move out of static item
  |                                 help: consider borrowing here: `&x`

error[E0507]: cannot move out of immutable static item `x`
 --> src/main.rs:3:33
  |
3 |     let _y = { static x: S = S; x };
  |                                 ^
  |                                 |
  |                                 cannot move out of immutable static item `x`
  |                                 cannot move

error: aborting due to 2 previous errors
```

It issues only a single error under either of AST-borrowck or pure NLL.

**Comment by pnkfelix** [comment]
This will be fixed by PR #55221

**Comment by matthewjasper** [comment]
Fixed in the latest nightly
